pkg/version/version.go

```diff
@@ -34,26 +30,32 @@ type CiliumVersion struct {
 	GoRuntimeVersion string
 	// Arch is the architecture where Cilium was compiled
 	Arch string
+	// BuildDate is the build time reference stored as string ISO 8601 formatted
+	BuildDate string
```

Comment:
From what I can tell, this is the git Author date, and not the build date. Did you intend for it to be the git date? If so, could we change the name to better reflect that?
If not, we'll have to inject the output of `date -u '+%Y-%m-%dT%H:%M:%S%z'` into the `Build` variable in Makefiles.def. We'd also have to parse it appropriately. To simplify parsing, it might easier to include a simple delimiter in the raw string (tabs might work, since they won't ruin the human readable version). We can then trivially split in this file, instead of constructing a more complex regex.

Reply:
@raybejjani  I changed the name for this one to AuthorDate. We can always add build date if you see some use cases.

Reply:
Excellent. Thank you. I've historically found build dates useful when the version info wasn't good enough (or when I worked on something that released multiple builds of the same thing without marking them). That's not the case here, I don't think.